### DISCUSSION TITLE
Potential Series X Port?

### DISCUSSION THREAD

**Discussion by ghost** [flyinghead/flycast#408] [Q&A] [answered]
Hey, would anyone be interested in porting Flycast to the Series consoles? It runs pretty well through Retroarch on the Series, but a standalone port would make playing these games a dream.

**Comment by flyinghead** [comment, marked as answer]
I don't have access to a Series X console so I'll need help for testing. I managed to make a .appx that installs and runs on windows 10. If anybody feels adventurous... (rename to .appx)

[flycast.zip](https://github.com/flyinghead/flycast/files/7572359/flycast.zip)

**Reply by ghost** [reply]
Awesome! Thanks so much. I’ll send this to some people! I really appreciate this!

**Reply by Gamr13** [reply]
Hi there! 

Do you perhaps have an Xbox One to test on? The Xbox One and Series consoles should be near identical in terms of what can and cannot run. I'll gladly give this a test on my Series X in the meantime.

Cheers!

**Reply by ghost** [reply]
Hi again! Emulators on the Xbox consoles require either Direct X 11 or 12. Would it be a possibility to add support for this in flycast? That being said, do you have any places we can donate to show our support and appreciation? Thanks again!

**Reply by flyinghead** [reply]
Right now Flycast uses ANGLE on this platform, which translates from Open GL ES 3.0 to DirectX 11. There's no plan for a native DirectX 11 renderer but we'll see how it goes.

**Reply by tunip3** [reply]
its working :0 👍

**Reply by flyinghead** [reply]
Seriously? Pics or it didn't happen

**Reply by Gamr13** [reply]
Here are my findings: (Flycast set to Game in dev mode, tested Xbox One S and Series X)

- Some menus are impossible to navigate without a keyboard (There's no way out of setting a directory, for example)
- Pressing B on the controller will immediately crash the app, in the menus or in a game
- The screen flashes white on Series X in the menus sometimes (Doesn't seem to exist on Xbox One S)
- Trying to activate the Menu from in game will crash the app
- A lot of screen flickering in-game (Series X only, it seems (May also apply to Series S, but I can't test that))

**Reply by flyinghead** [reply]
I noticed that activating the menu from in game does crash. This doesn't seem to happen in single-thread mode (Settings > Advanced > Multi-threaded emulation) but performance will suffer.

No idea why the B button would crash though. I didn't notice it on windows but I'll double-check. That would be the button to go back/exit in the UI when configuring a content directory for example.

**Comment by Gamr13** [comment]
> Seriously? Pics or it didn't happen

Trust me, can confirm it works, it's just super buggy, I've attached some images, and even managed to get a picture of the screen flashes.

![20211119_201431](https://user-images.githubusercontent.com/48205338/142686288-3060cee7-2e04-45a0-bcb2-f37dbb9d26da.jpg)
![Uploading 20211119_201447.jpg…]()
![20211119_201502](https://user-images.githubusercontent.com/48205338/142686296-dcb03af2-c137-46fe-8370-07fee80c8c0b.jpg)
![20211119_201518](https://user-images.githubusercontent.com/48205338/142686298-dee65c80-75da-464a-a040-ea169635a0c1.jpg)
![20211119_201423](https://user-images.githubusercontent.com/48205338/142686275-36f6b607-e5cb-4d06-b815-1812dc53abe6.jpg)

**Reply by flyinghead** [reply]
Thanks for testing and reporting guys! This is very encouraging for a first build.

**Reply by Gamr13** [reply]
We're happy to help. This is a big step for the community, the standalone port you have here performs substantially better than the RetroArch core, from our current findings. We appreciate you working on a standalone port.

**Reply by ghost** [reply]
Hey man, I also have been very happy testing some stuff on my Series X for fun. I can report a solid 60fps on Crazy Taxi with some occasional screen flashes. Hydro Thunder ran at a very consistent 30fps framerate with very stable visuals as well. Sadly, Daytona USA, while very solid in performance and in the menus, has extreme screen flashes when you are in-game, making it unplayable.

But I can concur with Gamr13 that this port is an immense improvement from the RA core also running under ANGLE on a Series X. I'm looking forward to what comes out of it @flyinghead. Thank you for this, I'm really grateful for this.

Thanks for reading.

**Comment by flyinghead** [comment]
To answer your question, you can find links to donate on the [main project page](/flyinghead/flycast) under "Sponsor this project"

**Comment by tunip3** [comment]
Do you know where we can find the source code for the port? @flyinghead

**Reply by flyinghead** [reply]
I haven't committed the changes just yet. I'm still trying to make an automated CI build, without much success so far...

**Comment by flyinghead** [comment]
I finally managed to get an automated build: https://github.com/flyinghead/flycast/actions/runs/1485057045
Can someone try it out?

I haven't signed the package this time but the crash when opening the menu in game should be fixed.

**Reply by dominater01** [reply]
works on pc

![Untitled](https://user-images.githubusercontent.com/8470028/142760096-3db840eb-c3bc-498f-a7f4-90b5d00e3ec0.png)

**Reply by flyinghead** [reply]
Not sure why you edited your comment but I take it that the package must be signed to be installed on xbox.

Thanks for testing.

**Comment by flyinghead** [comment]
New build, signed this time: https://github.com/flyinghead/flycast/actions/runs/1487062885

**Comment by dominater01** [comment]
> Not sure why you edited your comment but I take it that the package must be signed to be installed on xbox.
> 
> Thanks for testing.

because you just wanted it tested to see if it no longer crashes so i thought you knew it wouldnt install on xbox without it being signed so instead of just telling you it wouldnt install i just tested it on pc

**Comment by dominater01** [comment]
> New build, signed this time: https://github.com/flyinghead/flycast/actions/runs/1487062885

still crashes on xbox series s when opening that menu

**Comment by flyinghead** [comment]
This build fixes the B button closing the app (thanks @tunip3) :
https://github.com/flyinghead/flycast/actions/runs/1499381131
